apps/keyboard/js/views/key_view.js

```diff
@@ -12,6 +14,25 @@ function KeyView(target, options, viewManager) {
 
   if (target.isSpecialKey) {
     this.classNames.push('special-key');
+
+    // Make some special buttons to behave as keys for better
+    // interaction with screen readers. See bug#1212588 for
+    // discussion.
+    var specialKeysNotButtons = [
+      KeyEvent.DOM_VK_BACK_SPACE,
+      KeyEvent.DOM_VK_CAPS_LOCK,
+      KeyEvent.DOM_VK_ALT
+    ];
+
+    var willHaveKeyRole = target.keyCode &&
```

Comment:
I left the expression with a name just for clarity, but you are definitely right.